addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/LEDStateDTO.java

```diff
@@ -70,18 +75,18 @@ public static LEDStateDTO valueOf(int state, int program, int programSpeed, int
         OnOffType power = (state & 0x01) != 0 ? OnOffType.ON : OnOffType.OFF;
         float[] hsv = new float[3];
         Color.RGBtoHSB(red, green, blue, hsv);
-        long hue = (long) (hsv[0] * 360);
-        int saturation = (int) (hsv[1] * 100);
-        int brightness = (int) (hsv[2] * 100);
+        BigDecimal hue = BigDecimal.valueOf(hsv[0] * 360);
```

Comment:
Martin,

your assumption that hue is an integer is incorrect.
For example with the RGB values 128,4,200 the hue, saturation and brightness would be 
```
0 = 0.77210885
1 = 0.98
2 = 0.78431374
```

Within the constructor it get's multiplied by 360 but the getColor() method  later on in this class makes a division by 360 but based on integer. So you get:

0.77210885 8 360 = 277.959186

Then casting to int and dividing by 360 you get : 0.769444
Even if you properly round up or down, you don't get the exact hue value back and thus your RGB value might be off by one digit.

If you mean that the commit message should be updated, I can... I tried to be brief...